### PR TITLE
add OTLP Exporter customization

### PR DESCRIPTION
**Pull request by malafeev**
fix for #564

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,6 +78,9 @@ A simple wrapper for the OTLP exporter of opentelemetry-java.
 | ota.exporter=otlp (default)      | OTA_EXPORTER=otlp                | To select OpenTelemetry exporter (default)                           |
 | ota.exporter.jar                 | OTA_EXPORTER_JAR                 | Path to the exporter fat-jar that you want to use                    |
 | ota.exporter.otlp.endpoint       | OTA_EXPORTER_OTLP_ENDPOINT       | The OTLP endpoint to connect to.                                     |
+| ota.exporter.otlp.deadline.ms    | OTLP_DEADLINE_MS                 | Max waiting time for the collector to process each span batch        |
```
Comment (iNikem) [inline README.md:81]: Environment variable names are inconsistent with already existing ones. Please fix

File: auto-exporters/otlp/src/main/java/io/opentelemetry/auto/exporters/otlp/OtlpSpanExporterFactory.java [hunk 2]
```diff
@@ -15,23 +15,85 @@
...
     if (otlpEndpoint.isEmpty()) {
       throw new IllegalStateException("ota.exporter.otlp.endpoint is required");
     }
-    return OtlpGrpcSpanExporter.newBuilder()
-        .setChannel(ManagedChannelBuilder.forTarget(otlpEndpoint).usePlaintext().build())
-        .build();
+
+    final Builder builder = OtlpGrpcSpanExporter.newBuilder();
+
+    final long deadlineMs = config.getLong(OTLP_DEADLINE_MS, -1);
```
Comment (iNikem) [inline auto-exporters/otlp/src/main/java/io/opentelemetry/auto/exporters/otlp/OtlpSpanExporterFactory.java:53]: no new/changed tests for new functionality?

File: README.md (hunk 1 above)
Comment (malafeev) [reply README.md]: fixed

File: auto-exporters/otlp/src/main/java/io/opentelemetry/auto/exporters/otlp/OtlpSpanExporterFactory.java [hunk 3]
```diff
@@ -15,23 +15,85 @@
...
+    if (metadata != null) {
+      final Map<Key<String>, String> customMetadata = new HashMap<>();
+      for (String keyValueString : metadata.split(";")) {
+        final String[] keyValue = keyValueString.split("=");
+        if (keyValue.length == 2) {
+          customMetadata.put(Key.of(keyValue[0], ASCII_STRING_MARSHALLER), keyValue[1]);
+        }
+      }
+      if (!customMetadata.isEmpty()) {
+        managedChannelBuilder.intercept(
```
Comment (anuraaga) [inline auto-exporters/otlp/src/main/java/io/opentelemetry/auto/exporters/otlp/OtlpSpanExporterFactory.java:77]: Is it possible to use `MetadataUtils.newAttachHeadersInterceptor`?

**Comment by trask** [comment]
Hi @malafeev! Can you look at adding this configuration directly to the OTLP exporter? E.g. it already supports `OTEL_OTLP_SPAN_TIMEOUT` which maps to `deadlineMs`.

And check out #531, where we still need to transition this project over to the new(-ish) standard exporter configuration.

**Comment by malafeev** [comment]
@trask  in the OTLP exporter `ManagedChannel` is passed as parameter, but I need to configure it before it is built.
Parameters should be read from env variables/system properties and then build `ManagedChannel` based on them.
Although discussion should be probably moved to https://github.com/open-telemetry/opentelemetry-java

**Comment by trask** [comment]
let's see if we can upstream this to otel-java, it seems to fit in with general env-based exporter configuration, but if not for some reason then we can definitely add it here 👍

**Comment by malafeev** [comment]
ok, let me try

**Comment by trask** [comment]
@malafeev, do u mind if we convert this to draft, or close it, while we pursue upstreaming it to otel-java?

**Comment by malafeev** [comment]
@trask I cannot convert it to draft

**Comment by malafeev** [comment]
converted to draft

**Comment by malafeev** [comment]
created PR in opentelemetry-java: https://github.com/open-telemetry/opentelemetry-java/pull/1373

**Comment by malafeev** [comment]
not needed anymore because open-telemetry/opentelemetry-java#1373 was merged

**Pull request closed without merging**